packages/amplify-category-auth/src/provider-utils/awscloudformation/index.ts

```diff
@@ -1,46 +1,51 @@
-// TODO enable eslint after converting to TS
-/* eslint-disable */
-const inquirer = require('inquirer');
-const _ = require('lodash');
-const { stateManager, open } = require('amplify-cli-core');
-const { ensureEnvParamManager } = require('@aws-amplify/amplify-environment-parameters');
-const { getAuthResourceName } = require('../../utils/getAuthResourceName');
-const { copyCfnTemplate, saveResourceParameters } = require('./utils/synthesize-resources');
-const {
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable jsdoc/require-jsdoc */
+import inquirer from 'inquirer';
+import _ from 'lodash';
+import { stateManager, open } from 'amplify-cli-core';
+import { ensureEnvParamManager } from '@aws-amplify/amplify-environment-parameters';
+import { getAuthResourceName } from '../../utils/getAuthResourceName';
+import { copyCfnTemplate, saveResourceParameters } from './utils/synthesize-resources';
+import {
   ENV_SPECIFIC_PARAMS, AmplifyAdmin, UserPool, IdentityPool, BothPools, privateKeys,
-} = require('./constants');
-const { getAddAuthHandler, getUpdateAuthHandler } = require('./handlers/resource-handlers');
-const { getSupportedServices } = require('../supported-services');
-const { importResource, importedAuthEnvInit } = require('./import');
-
-const serviceQuestions = (context, defaultValuesFilename, stringMapsFilename, serviceWalkthroughFilename, serviceMetadata) => {
+} from './constants';
+import { getAddAuthHandler, getUpdateAuthHandler } from './handlers/resource-handlers';
+import { getSupportedServices } from '../supported-services';
+import { importResource, importedAuthEnvInit } from './import';
+
+export { importResource } from './import';
+
+const serviceQuestions = async (context:any,
+  defaultValuesFilename:any,
+  stringMapsFilename: any,
+  serviceWalkthroughFilename: any,
+  serviceMetadata: any): Promise<any> => {
   const serviceWalkthroughSrc = `${__dirname}/service-walkthroughs/${serviceWalkthroughFilename}`;
-  const { serviceWalkthrough } = require(serviceWalkthroughSrc);
+  const { serviceWalkthrough } = await import(serviceWalkthroughSrc);
   return serviceWalkthrough(context, defaultValuesFilename, stringMapsFilename, serviceMetadata);
 };
 
-const addResource = async (context, service, skipNextSteps = false) => {
+export const addResource = async (context: any, service: any): Promise<any> => {
```

Comment:
Do we have types for context and service? I recall we have `$TSContext` for example.
Is return type correct? should it be void?
Please revisit other places like this if this is valid concern.

Reply:
Imported $TSContext here and used it for all `context` parameters. On further investigation, it looks like the correct return type here is `Promise<string>`. I updated the method signature to reflect that.